test/https.cc

```diff
@@ -0,0 +1,138 @@
+#include "event-loop-test.h"
+#include "gtest/gtest.h"
+#include "stout/eventual.h"
+#include "stout/http.h"
+#include "stout/interrupt.h"
+#include "stout/lambda.h"
+#include "stout/terminal.h"
+
+namespace eventuals = stout::eventuals;
+
+using eventuals::EventLoop;
+using eventuals::Interrupt;
+using eventuals::Terminate;
+
+using eventuals::http::Get;
+using eventuals::http::Post;
+using eventuals::http::PostFields;
+
+class HTTPSTest : public EventLoopTest {};
+
+TEST_F(HTTPSTest, GetFailTimeout) {
```

Comment:
If I'm reading this correctly these are all the same tests as in `http.cc`? Rather than having to copy the code everywhere (which is a burden when you make changes in one and forget to do it in another) you can use the [Value Parameterized Tests](https://github.com/google/googletest/blob/master/docs/advanced.md#value-parameterized-tests) feature. Then you can do something like:
```
// On Windows we can't test with the 'https' scheme because we don't have SSL support.
#ifdef _WIN32
const char* schemes[] = {"http://"};
#else
const char* schemes[] = {"http://", "https://"};
#endif

INSTANTIATE_TEST_SUITE_P(Schemes, HttpTest, testing::ValuesIn(schemes));
```
And then inside each test you can do:
```
std::string scheme = GetParam();
auto e = Get(scheme + "example.com", ...
```